File: Laba2/Task5.cs
using Extensions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PZPK_Labas.Laba2
{
    public static class Task5
    {
        public static void Run()
        {
            Console.WriteLine("Завдання 5. У заданій матриці розмірності 6 на 4 визначити номер стовпця з мінімальною " +
                                "сумою елементів і величину цієї суми.");

            int[,] matrix = ArrayHelper.GenerateMatrix(4, 6, -50, 49);

            Console.WriteLine("Матриця:");
            ArrayHelper.PrintMatrix(matrix);

            CalculateRowWithWithMinSum(matrix);
        }

        private static void CalculateRowWithWithMinSum(int[,] matrix)
        {
            int minSum = int.MaxValue;
            int minSumColumnIndex = -1;
            for (int i = 0; i < matrix.GetLength(0); i++)
            {
                int sum = 0;
                for (int j = 0; j < matrix.GetLength(1); j++)
                    sum += matrix[i, j];
                if (sum < minSum)
                {
                    minSum = sum;
                    minSumColumnIndex = i;
                }
            }
            Console.WriteLine($"Номер стовпця з мінімальною сумою елементів: {minSumColumnIndex + 1}");
            Console.WriteLine($"Величина цієї суми: {minSum}");
        }
    }
}

File: Laba2-Python/Task5.py
import random
import ConsoleInput
import ArrayHelper

def Run():
    print("Завдання 5. У заданій матриці розмірності 6 на 4 визначити номер стовпця з мінімальною " +
          "сумою елементів і величину цієї суми.")

    matrix = ArrayHelper.GenerateMatrix(6, 4, -50, 49)

    print("Матриця:")
    ArrayHelper.PrintMatrix(matrix)

    calculate_column_with_min_sum(matrix)

def generate_matrix(rows, cols, min_value, max_value):
    matrix = []
    for _ in range(rows):
        row = [random.randint(min_value, max_value) for _ in range(cols)]
        matrix.append(row)
    return matrix

def print_matrix(matrix):
    for row in matrix:
        print(row)

def calculate_column_with_min_sum(matrix):
    min_sum = float('inf')
    min_sum_column_index = -1
    for j in range(len(matrix[0])):
        sum = 0
        for i in range(len(matrix)):
            sum += matrix[i][j]
        if sum < min_sum:
            min_sum = sum
            min_sum_column_index = j
    print("Номер стовпця з мінімальною сумою елементів:", min_sum_column_index + 1)
    print("Величина цієї суми:", min_sum)
